app/src/main/java/ro/code4/monitorizarevot/ui/splashscreen/SplashscreenActivity.kt

```diff
@@ -0,0 +1,43 @@
+package ro.code4.monitorizarevot.ui.splashscreen
+
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Bundle
+import org.koin.android.ext.android.inject
+import org.koin.android.viewmodel.ext.android.viewModel
+import ro.code4.monitorizarevot.R
+import ro.code4.monitorizarevot.helper.getToken
+import ro.code4.monitorizarevot.ui.base.BaseActivity
+import ro.code4.monitorizarevot.ui.login.LoginActivity
+import ro.code4.monitorizarevot.ui.main.MainActivity
+
+class SplashscreenActivity: BaseActivity<SplashscreenViewModel>()  {
+    private val sharedPreferences: SharedPreferences by inject()
+
+    override val layout: Int
+        get() = R.layout.activity_splashscreen
+    override val viewModel: SplashscreenViewModel by viewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Check if user is logged in or not
+        if (sharedPreferences.getToken() != null) {
+            startMainActivity()
+        } else {
+            startLoginActivity()
+        }
```

Comment:
Check the token in the viewmodel and then trigger the navigation using livedata

Reply:
I do not quite understand why this would be a better practice. Can you explain?

Reply:
The app follows an MVVM architecture. According to the Android [best practices](https://developer.android.com/jetpack/docs/guide) and the MVVM architecture the most important principle is the separation of the concerns. The UI classes should handle logic that handles UI and the ViewModel should handle business logic. Verifying that a user is logged should be in the business layer. Also, this condition may change in the future (e.g. checking if the token has expired, redirecting to login if the token expired, a network call checking if the token is still valid, etc.). Therefore you should move this condition in the ViewModel and trigger the navigation to the MainActivity/LoginActivity using a SingleLiveEvent.